YogaKit/Source/YGLayout.m

```diff
@@ -374,44 +384,87 @@ static CGFloat YGRoundPixelValue(CGFloat value)
   return roundf(value * scale) / scale;
 }
 
-static void YGApplyLayoutToViewHierarchy(UIView *view, BOOL preserveOrigin)
+static void YGApplyLayoutToHierarchy(id<YGLayoutEntity> entity, BOOL preserveOrigin, CGPoint offset)
 {
   NSCAssert([NSThread isMainThread], @"Framesetting should only be done on the main thread.");
 
-  const YGLayout *yoga = view.yoga;
+  const YGLayout *yoga = entity.yoga;
 
   if (!yoga.isIncludedInLayout) {
      return;
   }
 
   YGNodeRef node = yoga.node;
   const CGPoint topLeft = {
-    YGNodeLayoutGetLeft(node),
-    YGNodeLayoutGetTop(node),
+    YGNodeLayoutGetLeft(node) + offset.x,
+    YGNodeLayoutGetTop(node) + offset.y,
   };
 
-  const CGPoint bottomRight = {
-    topLeft.x + YGNodeLayoutGetWidth(node),
-    topLeft.y + YGNodeLayoutGetHeight(node),
-  };
-
-  const CGPoint origin = preserveOrigin ? view.frame.origin : CGPointZero;
-  view.frame = (CGRect) {
+  const CGPoint origin = preserveOrigin ? entity.frame.origin : CGPointZero;
+  entity.frame = (CGRect) {
     .origin = {
       .x = YGRoundPixelValue(topLeft.x + origin.x),
       .y = YGRoundPixelValue(topLeft.y + origin.y),
     },
     .size = {
-      .width = YGRoundPixelValue(bottomRight.x) - YGRoundPixelValue(topLeft.x),
-      .height = YGRoundPixelValue(bottomRight.y) - YGRoundPixelValue(topLeft.y),
+      .width = YGNodeLayoutGetWidth(node),
+      .height = YGNodeLayoutGetHeight(node),
     },
   };
 
   if (!yoga.isLeaf) {
-    for (NSUInteger i=0; i<view.subviews.count; i++) {
-      YGApplyLayoutToViewHierarchy(view.subviews[i], NO);
+    for (NSUInteger i=0; i<entity.subEntities.count; i++) {
+      YGApplyLayoutToHierarchy(entity.subEntities[i],
+                               NO,
+                               [entity conformsToProtocol:@protocol(UICoordinateSpace)] ? CGPointZero : entity.frame.origin);
     }
   }
 }
 
 @end
+
+@implementation YGLayoutContainer
+@synthesize frame=_frame;
+
+- (instancetype)init
+{
+  return [self initWithEntity:self];
+}
+- (instancetype)initWithEntity:(id<YGLayoutEntity>)entity
+{
+  if ((self = [super initWithEntity:entity])) {
+    self.isEnabled = YES;
+  }
+  return self;
+}
+
+- (YGLayout *)yoga
+{
+  return self;
+}
+
+- (CGSize)sizeThatFits:(CGSize)fitSize
+{
+  return CGSizeZero;
+}
+
+- (BOOL)isEnabled
+{
+  if (!super.isEnabled) {
```

Comment:
Hmmm tricky tricky.

Something about just doesn't seem right to me, if you set the property to `NO`, it will obey you, but, if it's set to `YES` it might still return `NO` do to side-effects. That seems a bit broken.

I have to take a deeper look at this later, it seems to me like this logic needs to live somewhere else.